### PR TITLE
plugins/oil: switch to mkNeovimPlugin + update options

### PR DESCRIPTION
**Pull request by GaetanLepage**

### CODE REVIEW

File: plugins/utils/oil.nix [hunk 1]
```diff
@@ -5,224 +5,135 @@
...
+    originalName = "oil.nvim";
+    defaultPackage = pkgs.vimPlugins.oil-nvim;
+
+    maintainers = [maintainers.GaetanLepage];
+
+    # TODO introduced 2024-03-18: remove 2024-05-18
+    deprecateExtraOptions = true;
+    optionsRenamedToSettings = [
+      "defaultFileExplorer"
+      "bufOptions"
```
Comment (traxys) [inline plugins/utils/oil.nix:20]: Does this work with the suboptions (as bufOptions is _not_ an option currently?)

File: plugins/utils/oil.nix [hunk 2]
```diff
@@ -5,224 +5,135 @@
...
+      "winOptions"
```
Comment (traxys) [inline plugins/utils/oil.nix:21]: Same

File: plugins/utils/oil.nix [hunk 3]
```diff
@@ -5,224 +5,135 @@
...
+      "keymaps"
+      "useDefaultKeymaps"
+      ["viewOptions" "showHidden"]
+      ["viewOptions" "isHiddenFile"]
+      ["viewOptions" "isAlwaysHidden"]
+      ["float" "padding"]
+      ["float" "maxWidth"]
+      ["float" "maxHeight"]
+      ["float" "border"]
+      ["float" "winOptions"]
```
Comment (traxys) [inline plugins/utils/oil.nix:35]: Same?

File: plugins/utils/oil.nix [hunk 4]
```diff
@@ -5,224 +5,135 @@
...
+      ["preview" "maxWidth"]
+      ["preview" "minWidth"]
+      ["preview" "width"]
+      ["preview" "maxHeight"]
+      ["preview" "minHeight"]
+      ["preview" "height"]
+      ["preview" "border"]
+      ["preview" "winOptions"]
```
Comment (traxys) [inline plugins/utils/oil.nix:43]: Same?

File: plugins/utils/oil.nix [hunk 5]
```diff
@@ -5,224 +5,135 @@
...
+      ["progress" "maxWidth"]
+      ["progress" "minWidth"]
+      ["progress" "width"]
+      ["progress" "maxHeight"]
+      ["progress" "minHeight"]
+      ["progress" "height"]
+      ["progress" "border"]
+      ["progress" "winOptions"]
```
Comment (traxys) [inline plugins/utils/oil.nix:51]: Same?

**Review by traxys — CHANGES_REQUESTED** [review]

File: plugins/utils/oil.nix (hunk 1 above)
Comment (GaetanLepage) [reply plugins/utils/oil.nix:20]: Yes, it is better to explicit each suboption. Now fixed.

File: plugins/utils/oil.nix [hunk 6]
```diff
@@ -5,224 +5,135 @@
...
+        Set to false if you still want to use netrw.
+      '';
 
-          format = formatOption;
-        };
+      columns = mkOption {
+        type = with types;
+          listOf
+          (
+            either
```
[the remark this answers is not in the document]
Comment (GaetanLepage) [reply plugins/utils/oil.nix:104]: According to the documentation, the columns are either a simple string or a lua table.
Now, we could let the user insert raw lua code for extra flexibility. Is this what you had in mind ?

[the remark this answers is not in the document]
Comment (GaetanLepage) [reply plugins/utils/oil.nix]: Ok ! Done

**Review by traxys — APPROVED** [review]

**Pull request merged**